packages/compass-generative-ai/src/tools-controller.ts

```diff
@@ -47,12 +148,161 @@ export class ToolsController {
       };
     }
 
+    if (this.runner.server && this.runner.server.tools.length === 0) {
+      registerTools(this.runner.server);
+    }
+
+    if (this.currentConnection && this.toolGroups.has('db-read')) {
+      const availableTools = this.runner.server?.tools ?? [];
+      for (const toolBase of availableTools) {
+        if (!readonlyTools.has(toolBase.name)) {
+          continue;
+        }
+
+        // TODO: the types here are all messaged up for many reasons, not least
+        // because somehing is up with it inferring that the z.object() is
+        // actually valid for inputSchema.
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+        tools[toolBase.name] = tool({
+          name: toolBase.name,
+          description: (toolBase as any).description as string, // TODO: protected
```

Comment:
On top of everything else a bunch of these props are now protected. Wondering if it would be worth having a method on a tool in mongodb-mcp-server that does this conversion to a vercel ai sdk tool for us?